lib/IRGen/GenCoverage.cpp

```diff
@@ -145,6 +145,7 @@ void IRGenModule::emitCoverageMapping() {
   auto *RecordsVal = llvm::ConstantArray::get(RecordsTy, FunctionRecords);
 
   // Create the coverage data header.
+  const unsigned NRecords = 0;
```

Comment:
What's up with this?

Reply:
The commit to llvm and clang that added the use of NRecords (dd1ea9de2e3e3ac80a620f71411a9a36449f2697) has it set to 0 in clang. Previous behavior was to use `FunctionRecords.size()` so this is an explicit behavior change. I was emulating clang's behavior here, but maybe we shouldn't be. Maybe @vedantk can weigh in here?

Reply:
Thanks for doing this. This doesn't look correct to me. I think this change "says" that there aren't any functions covered in the translation unit. But actually, if this PR gets master-next building, I'd be very grateful. Then I can actually start doing the work of integrating dd1ea9de2e3e3 properly. So, +1 from me, let's at least get to the point where some tests are failing.

Reply:
Actually, my last comment is incorrect. NRecords /is/ supposed to be 0, as the function records are no longer affixed to the coverage header. So this change is correct!